src/main/java/de/tum/in/www1/artemis/service/learningpath/LearningPathNavigationService.java

```diff
@@ -0,0 +1,142 @@
+package de.tum.in.www1.artemis.service.learningpath;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.LearningObject;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.competency.LearningPath;
+import de.tum.in.www1.artemis.service.LearningObjectService;
+import de.tum.in.www1.artemis.service.learningpath.LearningPathRecommendationService.RecommendationState;
+import de.tum.in.www1.artemis.web.rest.dto.competency.LearningPathNavigationDTO;
+import de.tum.in.www1.artemis.web.rest.dto.competency.LearningPathNavigationObjectDTO;
+import de.tum.in.www1.artemis.web.rest.dto.competency.LearningPathNavigationObjectDTO.LearningObjectType;
+import de.tum.in.www1.artemis.web.rest.dto.competency.LearningPathNavigationOverviewDto;
+
+/**
+ * Service for navigating through a learning path.
+ */
+@Profile(PROFILE_CORE)
+@Service
+public class LearningPathNavigationService {
+
+    private final LearningPathRecommendationService learningPathRecommendationService;
+
+    private final LearningObjectService learningObjectService;
+
+    public LearningPathNavigationService(LearningPathRecommendationService learningPathRecommendationService, LearningObjectService learningObjectService) {
+        this.learningPathRecommendationService = learningPathRecommendationService;
+        this.learningObjectService = learningObjectService;
+    }
+
+    /**
+     * Map learning objects to a navigation DTO.
+     *
+     * @param learningPathUser          the user of the learning path
+     * @param progress                  the progress of the learning path
+     * @param predecessorLearningObject the predecessor learning object
+     * @param currentLearningObject     the current learning object
+     * @param successorLearningObject   the successor learning object
+     * @return the navigation DTO
+     */
+    private LearningPathNavigationDTO mapLearningObjectsToNavigationDto(User learningPathUser, int progress, LearningObject predecessorLearningObject,
+            LearningObject currentLearningObject, LearningObject successorLearningObject) {
+        return new LearningPathNavigationDTO(LearningPathNavigationObjectDTO.of(predecessorLearningObject, learningPathUser),
+                LearningPathNavigationObjectDTO.of(currentLearningObject, learningPathUser), LearningPathNavigationObjectDTO.of(successorLearningObject, learningPathUser),
+                progress);
+    }
+
+    /**
+     * Get the navigation for the given learning path.
```

Comment:
this JavaDoc is not very helpful, at least for me. The text above is basically the same as when I read out the method. The param and return description only repeats the variable name.